smartdashboard/view_builders.py

```diff
@@ -466,3 +445,149 @@ def overview_builder(manifest: Manifest) -> OverviewView:
         ens_view = ens_builder(manifest)
 
     return OverviewView(exp_view, app_view, orc_view, ens_view)
+
+
+def db_telem_builder(manifest: Manifest) -> TelemetryView:
+    """Database Telemetry page to be rendered
+
+    This function organizes the views within
+    the Database Telemetry page.
+
+    :param manifest: Manifest of the Experiment
+    :type manifest: Manifest
+    :return: View of the DB Telemetry Page
+    :rtype: TelemetryView
+    """
+    st.header("Database Telemetry")
+
+    st.write("")
+
+    col1, _ = st.columns([6, 6])
+    with col1:
+        selected_orchestrator_tuple = st.selectbox(
+            "Select an orchestrator:",
+            manifest.orcs_with_run_ctx,
+            format_func=lambda tup: f"{tup[1].name}: Run {tup[0]}",
+        )
+
+    st.write("")
+
+    if selected_orchestrator_tuple is not None:
+        run_id, selected_orchestrator = selected_orchestrator_tuple
+    else:
+        selected_orchestrator = None
+
+    shards = selected_orchestrator.shards if selected_orchestrator else []
+
+    if selected_orchestrator:
+        st.subheader(f"{selected_orchestrator.name}: Run {run_id} Telemetry")
+    else:
+        st.subheader("No Orchestrator Selected")
+
+    st.write("")
+
+    ### Orchestrator Summary ###
+    orc_summary_view = orc_summary_builder(selected_orchestrator)
+    st.write("")
+
+    ### Memory ###
+    memory_view = memory_view_builder(shards)
+    st.write("")
+
+    ### Clients ###
+    client_view = client_view_builder(shards)
+    st.write("")
+
+    return TelemetryView(orc_summary_view, memory_view, client_view)
+
+
+def memory_view_builder(shards: t.List[Shard]) -> MemoryView:
+    """Memory section of Database Telemetry page to be rendered
+
+    :param shards: Shards of the selected Orchestrator
+    :type shards: t.List[Shard]
+    :return: View of the memory portion of the DB Telemetry page
+    :rtype: MemoryView
+    """
+    with st.expander(label="Memory"):
+        col1, col2 = st.columns([6, 6])
+        with col1:
+            view = MemoryView(shards[0] if shards else None)
+            shard = st.selectbox(
+                "Select a shard:",
+                shards,
+                format_func=lambda shard: shard.name,
+                key="memory_shard",
+            )
+            view.update_shard(shard)
+            view.memory_table_element = st.empty()
+        with col2:
+            st.write("")
+            st.write("")
+            st.write("")
+            view.memory_graph_element = st.empty()
+
+    return view
+
+
+def client_view_builder(shards: t.List[Shard]) -> ClientView:
```

Comment:
After our conversation, I changed `ClientView` to a frozen dataclass and implemented `client_view_builder` to return `ClientView(shard, client_table_element, client_graph_element)`. I'm going to leave this unresolved just in case!